lib/NetworkManager.js

```diff
@@ -94,6 +110,14 @@ class NetworkManager extends EventEmitter {
       this._handleRequestStart(request._requestId, event.interceptionId, event.redirectUrl, event.request);
       return;
     }
+
+    if (event.authChallenge) {
```

Comment:
could there be `redirectStatusCode` and `authChallendge`?

Reply:
The case were the server both sends an authentication challenge and a redirect location header should not happen and is not valid.  
But from what I understand authentication takes over redirect. To check chromium/puppeter's behavior I've forged a fake auth page that presents both auth challenge and redirect headers :
- if http status code is 401 or 407 (auth challenge), then event.redirectStatusCode is undefined
- if any other status code, then event.authChallenge is undefined

I'll still move the block before the redirect test just in case chromium changes their behavior one day.